scripts/data/generate_data.py

```diff
@@ -265,24 +265,49 @@ def generate_data(
             block = fetch_block_header(next_block_hash)
         elif mode == "full":
             block = fetch_block(next_block_hash)
+
             for tx in block["data"]:
                 for idx, output in enumerate(tx["outputs"]):
                     outpoint = (tx["txid"], idx)
                     utxo_set[outpoint] = output
 
+                    output["cached"] = False
+
+            for tx in block["data"]:
```

Comment:
You can do in a single loop, but the order should be the following:
- First you iterate over all transaction inputs and set `cached=True` if the outpoint is in the utxo_set
- Then you iterate over all transaction outputs and add them to the utxo_set